marconi-core/src/Marconi/Core/Transformer/WithStream/Streamable.hs

```diff
@@ -0,0 +1,133 @@
+module Marconi.Core.Transformer.WithStream.Streamable where
+
+import Control.Concurrent.STM (TBQueue, atomically, readTBQueue, writeTBQueue)
+import Control.Monad (unless)
+import Control.Monad.IO.Class (liftIO)
+import Data.Binary (Binary, decodeOrFail, encode)
+import Data.ByteString qualified as BS
+import Data.ByteString.Lazy qualified as BL
+import Marconi.Core.Type (Point, Timed)
+import Network.Socket (Socket)
+import Network.Socket.ByteString qualified as SBS
+import Network.Socket.ByteString.Lazy (sendAll)
+import Streaming (Of, Stream)
+import Streaming.Prelude (repeatM, yield)
+
+-- | Something that can have mapped events streamed to or from it
+class Streamable a r where
+  streamFrom :: a -> Stream (Of r) IO ()
+  streamTo :: (Timed (Point event) event -> r) -> a -> Timed (Point event) event -> IO ()
+
+{- | A streaming implementation using @Socket@. Here's an example of how
```

Comment:
If that's already implemented in the test module, I would just point the user to that test code in the test module.